core/src/main/java/org/geysermc/geyser/network/LoggingPacketHandler.java

```diff
@@ -39,18 +45,39 @@
  */
 public class LoggingPacketHandler implements BedrockPacketHandler {
     protected final GeyserImpl geyser;
-    protected final GeyserSession session;
+    /**
+     * Sessions have a client id number from 0-n and are looked up in this list by client id
+     */
+    @Getter
+    protected List<GeyserSession> sessions = new ArrayList<>(1);
 
     LoggingPacketHandler(GeyserImpl geyser, GeyserSession session) {
         this.geyser = geyser;
-        this.session = session;
+        this.sessions.add(session.getClientId(), session);
     }
 
     boolean defaultHandler(BedrockPacket packet) {
-        geyser.getLogger().debug("Handled packet: " + packet.getClass().getSimpleName());
+        geyser.getLogger().debug("Handled packet: " + packet.getClass().getSimpleName() + " for client " + packet.getClientId());
         return false;
     }
 
+    /**
+     * Get all the client sessions logged in via this device.
+     * This gets the sessions of the primary player, and any subsequent players logged in via splitscreen.
+     */
+    public List<GeyserSession> getSessions() {
+        return ImmutableList.<GeyserSession>builder() // builderWithExpectedSize is probably not a good idea yet as older Spigot builds probably won't have it.
```

Comment:
No need to make this immutable since there's no planned API access.

Reply:
Ended up removing this method - it wasn't really necessary